test/unittests/CandidateTest.php

```diff
@@ -582,6 +582,61 @@ public function testGetValidSubprojectsReturnsEmptyArray(): void
         $this->assertEquals($this->_candidate->getValidSubprojects(), array());
     }
 
+    /**
+     * Test getSubprojectForMostRecentVisit returns most recent visit's label
+     *
+     * @covers Candidate::getSubprojectForMostRecentVisit
+     * @return void
+     */
+    public function testGetSubprojectForMostRecentVisitReturnsMostRecentVisitLabel()
+    {
+        $subproject = array(
+            array(
+                'SubprojectID' => 1,
+                'title'        => 'testSubproject' 
+            )
+        );
+
+        $this->_setUpTestDoublesForSelectCandidate();
+
+        $this->_dbMock->expects($this->any())
+            ->method('pselect')
+            ->with($this->stringContains(
+                "SELECT SubprojectID, title FROM session
+                    LEFT JOIN subproject USING (SubprojectID)
+                    WHERE CandID=:cid AND Date_visit IS NOT NULL
```

Comment:
I think the unit tests might choke here on the prepared statements. `:cid` might get resolved to an actual value here making it so that the `stringContains()` check evalutes to false. It might work if you change this part to something more similar to what you had before, like just looking for `ORDER BY Date_visit DESC LIMIT 1` as this value cannot change.

Reply:
i'm trying to figure out how to make the test spit out what i want. i'm not quite sure how the testing works. if you look at `testGetValidSubprojectsReturnsAListOfSubprojects()` on line 528, i'm trying to do something similar to match the returned array. do you have any idea why doing something similar doesn't work?